src/app/client/src/app/modules/groups/components/member-actions/member-actions.component.ts

```diff
@@ -1,31 +1,71 @@
 import { ResourceService } from '@sunbird/shared';
-import { Component, OnInit, Input, EventEmitter, ViewChild, Output} from '@angular/core';
+import { Component, Input, EventEmitter, ViewChild, Output, OnDestroy } from '@angular/core';
 import * as _ from 'lodash-es';
 
+export interface IMemberActionData {
+  title: string;
+  description: string;
+  buttonText: string;
+  theme?: 'primary' | 'error';
+}
 @Component({
   selector: 'app-member-actions',
   templateUrl: './member-actions.component.html',
   styleUrls: ['./member-actions.component.scss']
 })
-export class MemberActionsComponent {
+export class MemberActionsComponent implements OnDestroy {
   @ViewChild('modal') modal;
+  @Input() action: string;
+  @Input() member: any;
+  @Output() modalClose = new EventEmitter<void>();
+  @Output() actionConfirm = new EventEmitter<any>();
 
-  @Input() modalName: string;
-  @Input() member: {};
-  @Output() modalClosed = new EventEmitter();
-  @Output() handleMember = new EventEmitter();
-  @Input() showModal;
-
+  memberActionData: IMemberActionData;
   constructor(public resourceService: ResourceService) {
-   }
+  }
+
+  ngOnInit() {
+    switch (this.action) {
```

Comment:
cases instead of action replace with below which are more readable 
1. promoteAsAdmin
2. RemoveFromGroup
3. DismissAsAdmin